### PR TITLE
Clarify supported compression methods for OTLP

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Fixes: https://github.com/open-telemetry/opentelemetry-specification/issues/1001#issuecomment-699020209

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]

File: specification/protocol/exporter.md [hunk 1]
```diff
@@ -13,9 +13,15 @@ The following configuration options MUST be available to configure the OTLP expo
...
 | Headers              | The headers associated with gRPC or HTTP requests.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_SPAN_HEADERS` `OTEL_EXPORTER_OTLP_METRIC_HEADERS` |
-| Compression          | Compression key for supported compression types. Supported compression: `gzip`| no compression              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
+| Compression          | Compression key for supported compression types. | No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
 | Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
 
+Supported values for OTEL_EXPORTER_OTLP_*COMPRESSION options:
+
+- If the value is missing then compressions is disabled.
+- `gzip` is the only supported compression method for `http` protocol.
+- `gzip` and `deflate` are supported compression methods for `grpc` protocol.
```
Comment (aabmass) [inline specification/protocol/exporter.md:23]: I took a deeper look at this and `deflate` isn't well supported across implementations (I don't see it in .NET or Java after a quick look). GRPC spec say to prefer [compression levels](https://github.com/grpc/grpc/blob/master/doc/compression.md#compression-levels-and-algorithms) over specific algorithms: 

> The set of supported algorithm is implementation dependent. In order to simplify the public API and to operate seamlessly across implementations (both in terms of languages but also different version of the same one), we introduce the idea of compression levels (such as "low", "medium", "high").
> 
> Levels map to concrete algorithms and/or their settings (such as "low" mapping to "gzip -3" and "high" mapping to "gzip -9") **automatically depending on what a peer is known to support.**

(some additional documentation [here](https://github.com/grpc/grpc/blob/master/doc/compression_cookbook.md#levels-vs-algorithms))

However it looks like "compression levels" are also not well supported across implementations (I do not see it in Python client for example). I'm not sure what is the best thing to put in spec here, maybe somebody with more GRPC knowledge could help.

Comment (tigrannajaryan) [reply specification/protocol/exporter.md:23]: @bogdandrutu are you familiar with the topic?

Comment (bogdandrutu) [reply specification/protocol/exporter.md:23]: Not at all, we can ask someone from gRPC to help maybe

Comment (tigrannajaryan) [reply specification/protocol/exporter.md]: OK, to avoid being blocked I reworded. "gzip" is the specified method. Others may be added by each language.

**Review by anuraaga — APPROVED** [review]

File: specification/protocol/exporter.md [hunk 2]
```diff
@@ -13,9 +13,14 @@ The following configuration options MUST be available to configure the OTLP expo
...
 | Headers              | The headers associated with gRPC or HTTP requests.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_SPAN_HEADERS` `OTEL_EXPORTER_OTLP_METRIC_HEADERS` |
-| Compression          | Compression key for supported compression types. Supported compression: `gzip`| no compression              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
+| Compression          | Compression key for supported compression types. | No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
 | Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
 
+Supported values for OTEL_EXPORTER_OTLP_*COMPRESSION options:
+
+- If the value is missing then compressions is disabled.
```
Comment (arminru) [inline specification/protocol/exporter.md:21]:
```suggestion
- If the value is missing, then compression is disabled.
```

**Review by arminru — APPROVED** [review]

File: specification/protocol/exporter.md [hunk 3]
```diff
@@ -13,9 +13,14 @@ The following configuration options MUST be available to configure the OTLP expo
...
 | Headers              | The headers associated with gRPC or HTTP requests.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_SPAN_HEADERS` `OTEL_EXPORTER_OTLP_METRIC_HEADERS` |
-| Compression          | Compression key for supported compression types. Supported compression: `gzip`| no compression              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
+| Compression          | Compression key for supported compression types. | No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
 | Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
 
+Supported values for OTEL_EXPORTER_OTLP_*COMPRESSION options:
```
Comment (aabmass) [inline specification/protocol/exporter.md:19]:
```suggestion
Supported values for `OTEL_EXPORTER_OTLP_*COMPRESSION` options:
```

**Review by aabmass — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

File: specification/protocol/exporter.md [hunk 4]
```diff
@@ -13,9 +13,14 @@ The following configuration options MUST be available to configure the OTLP expo
...
 | Headers              | Key-value pairs to be used as headers associated with gRPC or HTTP requests. See [Specifying headers](./exporter.md#specifying-headers-via-environment-variables) for more details.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_SPAN_HEADERS` `OTEL_EXPORTER_OTLP_METRIC_HEADERS` |
-| Compression          | Compression key for supported compression types. Supported compression: `gzip`| no compression              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
+| Compression          | Compression key for supported compression types. Supported compression: `gzip`| No value              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
 | Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 10s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
 
+Supported values for `OTEL_EXPORTER_OTLP_*COMPRESSION` options:
+
+- If the value is missing, then compressions is disabled.
```
Comment (bogdandrutu) [inline specification/protocol/exporter.md:21]: s/compressions/compression?

Comment (tigrannajaryan) [reply specification/protocol/exporter.md]: Fixed.

File: specification/protocol/exporter.md (hunk 3 above)
Comment (tigrannajaryan) [reply specification/protocol/exporter.md]: Fixed.

File: specification/protocol/exporter.md (hunk 2 above)
Comment (tigrannajaryan) [reply specification/protocol/exporter.md]: Fixed.

**Pull request merged**